controllers/auth.go

```diff
@@ -1225,7 +1225,7 @@ func (c *ApiController) HandleSamlLogin() {
 	relayState = url.QueryEscape(relayState)
 	samlResponse = url.QueryEscape(samlResponse)
 	targetUrl := fmt.Sprintf("%s?relayState=%s&samlResponse=%s",
-		slice[4], relayState, samlResponse)
+		slice[len(slice)-1], relayState, samlResponse)
```

Comment:
I'm gonna add my general comment here:)

Can you add test for the logic in 1219-1228 so that we had some idea what target url is generated here?